{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (5) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "# load in annovar annotated GTEx eQTLs\n",
    "gtex = pd.read_csv('gtex_eqtls_annotated_gts.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load in original gtex with gene id\n",
    "gtex_orig = pd.read_csv('/data5/16p12_RNA/GTEx/data/GTEx_Analysis_v7_eQTL/Cells_EBV-transformed_lymphocytes.v7.signif_variant_gene_pairs.txt.gz', sep='\\t')\n",
    "\n",
    "# add eGENE to gtex df\n",
    "gtex['gene_id'] = gtex_orig['gene_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load in gene name map\n",
    "mapp = pd.read_csv('../16p12.2_rnaseq_analysis/data/gene_names_mapping_new.tsv', sep='\\t')\n",
    "mapp = mapp.set_index('Name', drop=False)\n",
    "\n",
    "# keep only some columns\n",
    "cols = ['Name', 'Description']\n",
    "mapp = mapp[cols].copy()\n",
    "\n",
    "# add hgnc symbol to gtex df\n",
    "\n",
    "gtex['gene'] = gtex['gene_id'].apply(lambda s: mapp.loc[s, 'Description'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load in annotations\n",
    "anno = pd.read_csv('../16p12.2_rnaseq_analysis/outlier_expression_analysis/gene_annotations.txt', sep='\\t')\n",
    "\n",
    "# keep only some columns\n",
    "cols = ['#HGNCsymbol', 'pLI', 'pLI_%', 'BrainExpressed_Kang2011', 'Developmental_delay_DDD',\n",
    "       'DBD', 'DDDG2P', 'SFARI_Gene', 'Purcell_Schiz', 'NDD_all', 'OMIM_phenotype']\n",
    "\n",
    "anno = anno[cols].copy()\n",
    "\n",
    "# set index\n",
    "anno = anno.set_index('#HGNCsymbol', drop=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make list of genes with annotations\n",
    "hgnc_symbols = list(anno.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_pli(gene):\n",
    "    if gene in hgnc_symbols:\n",
    "        pli = anno.loc[gene, 'pLI']\n",
    "        if pli == '.':\n",
    "            return False\n",
    "        pli = float(pli)\n",
    "        if pli > 0.9:\n",
    "            return True\n",
    "    return False\n",
    "\n",
    "mapp['pLI'] = mapp['Description'].apply(get_pli)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def is_brain_expressed(gene):\n",
    "    if gene in hgnc_symbols:\n",
    "        val = anno.loc[gene, 'BrainExpressed_Kang2011']\n",
    "        if val == '1':\n",
    "            return True\n",
    "    return False\n",
    "\n",
    "mapp['BrainExpressed_Kang2011'] = mapp['Description'].apply(is_brain_expressed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def is_disorder_gene(gene):\n",
    "    if gene in hgnc_symbols:\n",
    "        val = anno.loc[gene, 'Developmental_delay_DDD']\n",
    "        if val == '1':\n",
    "            return True\n",
    "    return False\n",
    "\n",
    "mapp['Developmental_delay_DDD'] = mapp['Description'].apply(is_disorder_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "def is_disorder_gene(gene):\n",
    "    if gene in hgnc_symbols:\n",
    "        val = anno.loc[gene, 'DBD']\n",
    "        if val != '.':\n",
    "            return True\n",
    "    return False\n",
    "\n",
    "mapp['DBD'] = mapp['Description'].apply(is_disorder_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "def is_disorder_gene(gene):\n",
    "    if gene in hgnc_symbols:\n",
    "        val = anno.loc[gene, 'DDDG2P']\n",
    "        if val == '1':\n",
    "            return True\n",
    "    return False\n",
    "\n",
    "mapp['DDDG2P'] = mapp['Description'].apply(is_disorder_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "def is_disorder_gene(gene):\n",
    "    if gene in hgnc_symbols:\n",
    "        val = anno.loc[gene, 'SFARI_Gene']\n",
    "        if val != '.':\n",
    "            return True\n",
    "    return False\n",
    "\n",
    "mapp['SFARI_Gene'] = mapp['Description'].apply(is_disorder_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "def is_disorder_gene(gene):\n",
    "    if gene in hgnc_symbols:\n",
    "        val = anno.loc[gene, 'Purcell_Schiz']\n",
    "        if val == '1':\n",
    "            return True\n",
    "    return False\n",
    "\n",
    "mapp['Purcell_Schiz'] = mapp['Description'].apply(is_disorder_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cols = ['pLI', 'BrainExpressed_Kang2011',\n",
    "       'Developmental_delay_DDD', 'DBD', 'DDDG2P', 'SFARI_Gene',\n",
    "       'Purcell_Schiz']\n",
    "for col in cols:\n",
    "    gtex[col] = gtex['gene_id'].apply(lambda s: mapp.loc[s, col])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "samples = [s for s in gtex.columns if s.startswith('SG')]\n",
    "\n",
    "# create dict to store values\n",
    "d = pd.DataFrame(index=samples)\n",
    "d['all'] = 0\n",
    "for col in cols:\n",
    "    d[col] = 0\n",
    "\n",
    "# iterate through all gtex eqtls\n",
    "for i, row in gtex.iterrows():\n",
    "    # get alt allele defined by gtex\n",
    "    alt = row['Alt']\n",
    "    \n",
    "    # get frequency of alternat allele\n",
    "    freq_alt = row['gnomAD_genome_ALL']\n",
    "    if freq_alt == '.':\n",
    "        freq_alt = 0\n",
    "    else:\n",
    "        freq_alt = float(freq_alt)\n",
    "    \n",
    "    # get allele number of alternate allele\n",
    "    vcf_alt = row['vcf_alt']\n",
    "    vcf_alt = vcf_alt.split(',')\n",
    "    alt_number = 0\n",
    "    for j in range(len(vcf_alt)):\n",
    "        allele = vcf_alt[j]\n",
    "        if allele == alt:\n",
    "            alt_number = j + 1\n",
    "\n",
    "    # if alt allele not in vcf then skip\n",
    "    if alt_number == 0:\n",
    "        continue\n",
    "    \n",
    "    # is alt of ref the minor allele?\n",
    "    if freq_alt < 0.5:\n",
    "        maf_number = str(alt_number)\n",
    "    else:\n",
    "        maf_number = str(0)\n",
    "    \n",
    "    # for each sample, does it have the minor allele\n",
    "    for samp in samples:\n",
    "        gt = row[samp]\n",
    "        has_minor_eqtl = maf_number in gt\n",
    "        if has_minor_eqtl:\n",
    "            d.at[samp, 'all'] = d.at[samp, 'all'] + 1\n",
    "            \n",
    "            # also add to disease gene sets\n",
    "            for col in cols:\n",
    "                if row[col]:\n",
    "                    d.at[samp, col] = d.at[samp, col] + 1\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "d.index.name = 'sample'\n",
    "d.to_csv('minor_allele_summary_disease.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all</th>\n",
       "      <th>pLI</th>\n",
       "      <th>BrainExpressed_Kang2011</th>\n",
       "      <th>Developmental_delay_DDD</th>\n",
       "      <th>DBD</th>\n",
       "      <th>DDDG2P</th>\n",
       "      <th>SFARI_Gene</th>\n",
       "      <th>Purcell_Schiz</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SG001</th>\n",
       "      <td>104471</td>\n",
       "      <td>4936</td>\n",
       "      <td>52433</td>\n",
       "      <td>2418</td>\n",
       "      <td>1439</td>\n",
       "      <td>8117</td>\n",
       "      <td>1417</td>\n",
       "      <td>6513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG003</th>\n",
       "      <td>104984</td>\n",
       "      <td>5015</td>\n",
       "      <td>50592</td>\n",
       "      <td>2438</td>\n",
       "      <td>1453</td>\n",
       "      <td>7552</td>\n",
       "      <td>1385</td>\n",
       "      <td>7423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG006</th>\n",
       "      <td>101937</td>\n",
       "      <td>5146</td>\n",
       "      <td>51030</td>\n",
       "      <td>2224</td>\n",
       "      <td>1348</td>\n",
       "      <td>7097</td>\n",
       "      <td>1311</td>\n",
       "      <td>6416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG007</th>\n",
       "      <td>102677</td>\n",
       "      <td>5147</td>\n",
       "      <td>50787</td>\n",
       "      <td>2198</td>\n",
       "      <td>1981</td>\n",
       "      <td>7592</td>\n",
       "      <td>2056</td>\n",
       "      <td>6582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG002</th>\n",
       "      <td>106468</td>\n",
       "      <td>5611</td>\n",
       "      <td>50768</td>\n",
       "      <td>2477</td>\n",
       "      <td>1731</td>\n",
       "      <td>8283</td>\n",
       "      <td>1488</td>\n",
       "      <td>6810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG011</th>\n",
       "      <td>108101</td>\n",
       "      <td>4314</td>\n",
       "      <td>51752</td>\n",
       "      <td>2452</td>\n",
       "      <td>1651</td>\n",
       "      <td>7965</td>\n",
       "      <td>1281</td>\n",
       "      <td>6294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG025</th>\n",
       "      <td>101238</td>\n",
       "      <td>5150</td>\n",
       "      <td>48417</td>\n",
       "      <td>2149</td>\n",
       "      <td>1332</td>\n",
       "      <td>7526</td>\n",
       "      <td>2034</td>\n",
       "      <td>7260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG022</th>\n",
       "      <td>105410</td>\n",
       "      <td>5202</td>\n",
       "      <td>49680</td>\n",
       "      <td>2325</td>\n",
       "      <td>2407</td>\n",
       "      <td>7899</td>\n",
       "      <td>1697</td>\n",
       "      <td>6479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG024</th>\n",
       "      <td>104920</td>\n",
       "      <td>5292</td>\n",
       "      <td>49855</td>\n",
       "      <td>2192</td>\n",
       "      <td>2656</td>\n",
       "      <td>8041</td>\n",
       "      <td>1328</td>\n",
       "      <td>6883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG021</th>\n",
       "      <td>104531</td>\n",
       "      <td>5821</td>\n",
       "      <td>50708</td>\n",
       "      <td>2038</td>\n",
       "      <td>2508</td>\n",
       "      <td>7224</td>\n",
       "      <td>2260</td>\n",
       "      <td>7574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG023</th>\n",
       "      <td>100409</td>\n",
       "      <td>5653</td>\n",
       "      <td>48174</td>\n",
       "      <td>2170</td>\n",
       "      <td>2707</td>\n",
       "      <td>7689</td>\n",
       "      <td>2090</td>\n",
       "      <td>6857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG026</th>\n",
       "      <td>104680</td>\n",
       "      <td>4939</td>\n",
       "      <td>50727</td>\n",
       "      <td>2838</td>\n",
       "      <td>2355</td>\n",
       "      <td>8709</td>\n",
       "      <td>1503</td>\n",
       "      <td>5509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG155</th>\n",
       "      <td>104746</td>\n",
       "      <td>5400</td>\n",
       "      <td>50561</td>\n",
       "      <td>2610</td>\n",
       "      <td>2385</td>\n",
       "      <td>7712</td>\n",
       "      <td>1392</td>\n",
       "      <td>7281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG030</th>\n",
       "      <td>102977</td>\n",
       "      <td>4976</td>\n",
       "      <td>50733</td>\n",
       "      <td>2548</td>\n",
       "      <td>2576</td>\n",
       "      <td>7780</td>\n",
       "      <td>1325</td>\n",
       "      <td>6216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG027</th>\n",
       "      <td>97612</td>\n",
       "      <td>5367</td>\n",
       "      <td>47838</td>\n",
       "      <td>2284</td>\n",
       "      <td>2458</td>\n",
       "      <td>7745</td>\n",
       "      <td>1562</td>\n",
       "      <td>6662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG031</th>\n",
       "      <td>103314</td>\n",
       "      <td>4993</td>\n",
       "      <td>49339</td>\n",
       "      <td>2190</td>\n",
       "      <td>1192</td>\n",
       "      <td>6778</td>\n",
       "      <td>1447</td>\n",
       "      <td>6151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG069</th>\n",
       "      <td>107447</td>\n",
       "      <td>5457</td>\n",
       "      <td>51757</td>\n",
       "      <td>2513</td>\n",
       "      <td>2215</td>\n",
       "      <td>7382</td>\n",
       "      <td>1657</td>\n",
       "      <td>6535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG041</th>\n",
       "      <td>103061</td>\n",
       "      <td>5182</td>\n",
       "      <td>50460</td>\n",
       "      <td>2599</td>\n",
       "      <td>1466</td>\n",
       "      <td>7943</td>\n",
       "      <td>1144</td>\n",
       "      <td>7088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG042</th>\n",
       "      <td>103185</td>\n",
       "      <td>5287</td>\n",
       "      <td>48588</td>\n",
       "      <td>2797</td>\n",
       "      <td>1639</td>\n",
       "      <td>8216</td>\n",
       "      <td>1095</td>\n",
       "      <td>6307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG043</th>\n",
       "      <td>106854</td>\n",
       "      <td>5864</td>\n",
       "      <td>51252</td>\n",
       "      <td>2623</td>\n",
       "      <td>2448</td>\n",
       "      <td>7543</td>\n",
       "      <td>1176</td>\n",
       "      <td>7725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG044</th>\n",
       "      <td>101646</td>\n",
       "      <td>5204</td>\n",
       "      <td>51125</td>\n",
       "      <td>2607</td>\n",
       "      <td>1603</td>\n",
       "      <td>7847</td>\n",
       "      <td>1650</td>\n",
       "      <td>6685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG045</th>\n",
       "      <td>104641</td>\n",
       "      <td>4925</td>\n",
       "      <td>51128</td>\n",
       "      <td>3035</td>\n",
       "      <td>1553</td>\n",
       "      <td>7938</td>\n",
       "      <td>1676</td>\n",
       "      <td>6450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG046</th>\n",
       "      <td>105343</td>\n",
       "      <td>4818</td>\n",
       "      <td>52570</td>\n",
       "      <td>2601</td>\n",
       "      <td>1843</td>\n",
       "      <td>7865</td>\n",
       "      <td>1280</td>\n",
       "      <td>6026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG037</th>\n",
       "      <td>105158</td>\n",
       "      <td>5564</td>\n",
       "      <td>54188</td>\n",
       "      <td>2640</td>\n",
       "      <td>1975</td>\n",
       "      <td>8169</td>\n",
       "      <td>1533</td>\n",
       "      <td>6870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG038</th>\n",
       "      <td>105462</td>\n",
       "      <td>4525</td>\n",
       "      <td>54615</td>\n",
       "      <td>2590</td>\n",
       "      <td>2244</td>\n",
       "      <td>7867</td>\n",
       "      <td>1673</td>\n",
       "      <td>6666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG039</th>\n",
       "      <td>108291</td>\n",
       "      <td>5203</td>\n",
       "      <td>53945</td>\n",
       "      <td>2906</td>\n",
       "      <td>2340</td>\n",
       "      <td>7765</td>\n",
       "      <td>1871</td>\n",
       "      <td>6429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG040</th>\n",
       "      <td>106969</td>\n",
       "      <td>5368</td>\n",
       "      <td>53103</td>\n",
       "      <td>2621</td>\n",
       "      <td>1590</td>\n",
       "      <td>8545</td>\n",
       "      <td>1619</td>\n",
       "      <td>6678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG151</th>\n",
       "      <td>102306</td>\n",
       "      <td>5495</td>\n",
       "      <td>51617</td>\n",
       "      <td>2483</td>\n",
       "      <td>2900</td>\n",
       "      <td>8633</td>\n",
       "      <td>1586</td>\n",
       "      <td>7475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG149</th>\n",
       "      <td>102862</td>\n",
       "      <td>5644</td>\n",
       "      <td>48938</td>\n",
       "      <td>2695</td>\n",
       "      <td>1262</td>\n",
       "      <td>8497</td>\n",
       "      <td>1705</td>\n",
       "      <td>7280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG148</th>\n",
       "      <td>104843</td>\n",
       "      <td>6377</td>\n",
       "      <td>50253</td>\n",
       "      <td>2811</td>\n",
       "      <td>2664</td>\n",
       "      <td>8782</td>\n",
       "      <td>1762</td>\n",
       "      <td>7702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG152</th>\n",
       "      <td>105810</td>\n",
       "      <td>5449</td>\n",
       "      <td>48254</td>\n",
       "      <td>2706</td>\n",
       "      <td>965</td>\n",
       "      <td>7139</td>\n",
       "      <td>1328</td>\n",
       "      <td>6289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SG150</th>\n",
       "      <td>98508</td>\n",
       "      <td>5365</td>\n",
       "      <td>48766</td>\n",
       "      <td>2649</td>\n",
       "      <td>1459</td>\n",
       "      <td>7799</td>\n",
       "      <td>1751</td>\n",
       "      <td>6883</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           all   pLI  BrainExpressed_Kang2011  Developmental_delay_DDD   DBD  \\\n",
       "sample                                                                         \n",
       "SG001   104471  4936                    52433                     2418  1439   \n",
       "SG003   104984  5015                    50592                     2438  1453   \n",
       "SG006   101937  5146                    51030                     2224  1348   \n",
       "SG007   102677  5147                    50787                     2198  1981   \n",
       "SG002   106468  5611                    50768                     2477  1731   \n",
       "SG011   108101  4314                    51752                     2452  1651   \n",
       "SG025   101238  5150                    48417                     2149  1332   \n",
       "SG022   105410  5202                    49680                     2325  2407   \n",
       "SG024   104920  5292                    49855                     2192  2656   \n",
       "SG021   104531  5821                    50708                     2038  2508   \n",
       "SG023   100409  5653                    48174                     2170  2707   \n",
       "SG026   104680  4939                    50727                     2838  2355   \n",
       "SG155   104746  5400                    50561                     2610  2385   \n",
       "SG030   102977  4976                    50733                     2548  2576   \n",
       "SG027    97612  5367                    47838                     2284  2458   \n",
       "SG031   103314  4993                    49339                     2190  1192   \n",
       "SG069   107447  5457                    51757                     2513  2215   \n",
       "SG041   103061  5182                    50460                     2599  1466   \n",
       "SG042   103185  5287                    48588                     2797  1639   \n",
       "SG043   106854  5864                    51252                     2623  2448   \n",
       "SG044   101646  5204                    51125                     2607  1603   \n",
       "SG045   104641  4925                    51128                     3035  1553   \n",
       "SG046   105343  4818                    52570                     2601  1843   \n",
       "SG037   105158  5564                    54188                     2640  1975   \n",
       "SG038   105462  4525                    54615                     2590  2244   \n",
       "SG039   108291  5203                    53945                     2906  2340   \n",
       "SG040   106969  5368                    53103                     2621  1590   \n",
       "SG151   102306  5495                    51617                     2483  2900   \n",
       "SG149   102862  5644                    48938                     2695  1262   \n",
       "SG148   104843  6377                    50253                     2811  2664   \n",
       "SG152   105810  5449                    48254                     2706   965   \n",
       "SG150    98508  5365                    48766                     2649  1459   \n",
       "\n",
       "        DDDG2P  SFARI_Gene  Purcell_Schiz  \n",
       "sample                                     \n",
       "SG001     8117        1417           6513  \n",
       "SG003     7552        1385           7423  \n",
       "SG006     7097        1311           6416  \n",
       "SG007     7592        2056           6582  \n",
       "SG002     8283        1488           6810  \n",
       "SG011     7965        1281           6294  \n",
       "SG025     7526        2034           7260  \n",
       "SG022     7899        1697           6479  \n",
       "SG024     8041        1328           6883  \n",
       "SG021     7224        2260           7574  \n",
       "SG023     7689        2090           6857  \n",
       "SG026     8709        1503           5509  \n",
       "SG155     7712        1392           7281  \n",
       "SG030     7780        1325           6216  \n",
       "SG027     7745        1562           6662  \n",
       "SG031     6778        1447           6151  \n",
       "SG069     7382        1657           6535  \n",
       "SG041     7943        1144           7088  \n",
       "SG042     8216        1095           6307  \n",
       "SG043     7543        1176           7725  \n",
       "SG044     7847        1650           6685  \n",
       "SG045     7938        1676           6450  \n",
       "SG046     7865        1280           6026  \n",
       "SG037     8169        1533           6870  \n",
       "SG038     7867        1673           6666  \n",
       "SG039     7765        1871           6429  \n",
       "SG040     8545        1619           6678  \n",
       "SG151     8633        1586           7475  \n",
       "SG149     8497        1705           7280  \n",
       "SG148     8782        1762           7702  \n",
       "SG152     7139        1328           6289  \n",
       "SG150     7799        1751           6883  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
